stdlib/asm/math/ext2.masm

```diff
@@ -0,0 +1,84 @@
+# Given a stack with initial configuration given by [a1,a0,b1,b0,...] where a = (a0,a1) and
+# b = (b0,b1) represent elements in the extension field of degree 2, the procedure outputs the 
+# product c = (c1,c0) where c0 = a0b0 - 2(a1b1) and c1 = (a0 + a1)(b0 + b1) - a0b0
+export.mul_ext_2
+    dupw            #[a1,a0,b1,b0,a1,a0,b1,b0,...]
+    swap.3          #[b0,a0,b1,a1,a1,a0,b1,b0,...]
+    mul             #[b0a0,b1,a1,a1,a0,b1,b0,...]
+    dup             #[b0a0,b0a0,b1,a1,a1,a0,b1,b0,...]
+    movdn.7         #[b0a0,b1,a1,a1,a0,b1,b0,b0a0,...]
+    movdn.2         #[b1,a1,b0a0,a1,a0,b1,b0,b0a0,...]
+    mul.2           #[2b1,a1,b0a0,a1,a0,b1,b0,b0a0,...]
+    mul             #[2b1a1,b0a0,a1,a0,b1,b0,b0a0,...]
+    sub             #[b0a0-2b1a1,a1,a0,b1,b0,b0a0,...]
+    movdn.5         #[a1,a0,b1,b0,b0a0,b0a0-2b1a1,...]
+    add             #[a1+a0,b1,b0,b0a0,b0a0-2b1a1,...]
+    swap.2          #[b0,b1,a1+a0,b0a0,b0a0-2b1a1,...]
+    add             #[b0+b1,a1+a0,b0a0,b0a0-2b1a1,...]
+    mul             #[(b0+b1)(a1+a0),b0a0,b0a0-2b1a1,...]
+    swap            #[b0a0,(b0+b1)(a1+a0),b0a0-2b1a1,...]
+    sub             #[(b0+b1)(a1+a0)-b0a0,b0a0-2b1a1,...]
+end
+
+# Given a stack with initial configuration given by [x,a1,a0,...] where a = (a0,a1) is an element
+# in the field extension and x is an element of the base field, this procedure computes the multiplication
+# of x, when looked at as (x,0), with a in the extension field. The output is [xa1,xa0,...]
+export.mul_base
+    dup         #[x,x,a1,a0,...]
+    movdn.3     #[x,a1,a0,x,...]
+    mul         #[xa1,a0,x,...]
+    movdn.2     #[a0,x,xa1,...]
+    mul         #[xa0,xa1,...]
+    swap        #[xa1,xa0,...]
+end
+
+# Given a stack with initial configuration given by [a1,a0,...] where a = (a0,a1) is an element
+# in the field extension, calculate the result of applying the frobenius map to a
+# input: [a1,a0,...]
+# output: [-a1,a0+a1,...]
+export.frob
+    dup         #[a1,a1,a0,...]
+    movup.2     #[a0,a1,a1,...]
+    add         #[a0+a1,a1,...]
+    swap        #[a1,a0+a1,...]
+    neg         #[-a1,a0+a1,...]
+end
+
+# Given a stack with initial configuration given by [a1,a0,...] where a = (a0,a1) is an element
+# in the field extension, calculate the inverse of a.
+export.inv_ext
```

Comment:
Similar comment here and in other places: we can probably drop the `ext` suffix.